### PR TITLE
[MRG] ntp rework managment

### PR DESCRIPTION
**Pull request by bluegenes**
rework project management section to focus on workflow-based project management

### CODE REVIEW

File: content/04.workflows-and-software.md [hunk 1]
```diff
@@ -89,7 +88,8 @@ This ensures that an environment is completely reproducible, and is common for p
...
-also specifies which Conda channels to search and download programs from. When specified in a YAML file, conda environments are easily transferable between computers and operating systems. Further, because the version of each package installed in an environment is recorded, workflow reproducibility is enhanced. Although portions of conda may be superceded by alternative solutions [@url:https://github.com/QuantStack/mamba], this model of software installation and management will likely remain.](images/conda_figure.svg){#fig:conda_figure}
+also specifies which Conda channels to search and download programs from. When specified in a YAML file, conda environments are easily transferable between computers and operating systems. Further, because the version of each package installed in an environment is recorded, workflow reproducibility is enhanced. Although portions of conda may be superceded by alternative solutions [@url:https://github.com/QuantStack/mamba], this model of software installation and management will likely
+remain.](images/conda_figure.svg){#fig:conda_figure height=5in}
```
Comment (bluegenes) [inline content/04.workflows-and-software.md]:
```suggestion
remain.](images/conda_figure.svg){#fig:conda_figure height=8in}
```

File: content/05.project-management.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+**Start with working code**
+The best way to develop your own workflow analysis is to start from working code: either from the workflow tutorials or code sharing websites like GitHub, GitLab, and Bitbucket.
+If a workflow is available through Binder, you can test and experiment with workflow modification on Binder's cloud system, without needing to install a workflow manager or software management tool.
+When you're building a workflow for the first time, choose code in official repositories or reasonably well-used that provides sample data that can be run to verify the analysis works.
+The "Workflows and Software Management" Section links to a number of official repositories containing tutorials and example biological analysis workflows.
+
+**Test with subsampled data**
+After verifying your chosen example workflow is functional, try running it with your own data or some public data related to your species or condition of interest (see **Table {@tbl: seq_resources}**)
```
Comment (bluegenes) [inline content/05.project-management.md]:
```suggestion
After verifying your chosen example workflow is functional, try running it with your own data or some public data related to your species or condition of interest (see **Table {@tbl:seq_resources}**)
```

**Comment by bluegenes** [comment]
FYI - scaling /computational approximations is not gone, just moved to next section!

**Comment by bluegenes** [comment]
This PR focuses on reworking "project management" and "data and resource management" sections to be focused on workflow-enabled dib.

Abstract, Intro, Conclusion, Author Summary are still in stub/draft form, but slightly reworked and cleaned up.

File: content/01.abstract.md [hunk 3]
```diff
@@ -1,21 +1,9 @@
...
-
-The majority of this manuscript will covers understanding how to conduct computational analyses on sequencing data, 
-Except for data acquisition, the tools and guidelines presented below apply to either novel or publicly-available data.  
-
+As both sequencing technologies and data have proliferated, the bottleneck of biological sequence analysis has shifted from data generation to analysis.
+Fortunately, analysis tools and techniques have evolved to cope with this ever-increasing flood of data.
+Reliable and user-friendly workflow systems and software management have emerged to facilitate interrogation of many thousands of samples.
+For fundamental steps such as quality control, standardized protocols are now available meaning researchers can spend less time rewriting common analyses and more time examining the biological intricacies of their data.
+In cases where the data are too large for even high-performance computing environments, a series of tools have emerged that are capable of using small, representative subsets of massive datasets to produce comparable results.
```
Comment (taylorreiter) [inline content/01.abstract.md:7]: do we think something that gets one paragraph of air time in the paper is important enough to have a whole sentence in the abstract? Should we put more emphasis on something else?

File: content/02.author-summary.md [hunk 4]
```diff
@@ -1,7 +1,4 @@
 ## Author Summary {.page_break_before}
 
-In this paper, we present our guide for biological sequence data analysis, developed through our own teaching, training and analysis. We recognize that this is currently biased towards our own use casea nd experiences, but we hope to engage in robust discussion with the open source community in order in include the best set of practices.
-
+In this paper, we present our guide for workflow-enabled biological sequence data analysis, developed through our own teaching, training and analysis. We recognize that this is currently biased towards our own use case and experiences, but we hope to engage in robust discussion with the open source community to build a comprehensive resource.
```
Comment (taylorreiter) [inline content/02.author-summary.md]:
```suggestion
In this paper, we present our guide for workflow-enabled biological sequence data analysis, developed through our own teaching, training and analysis. We recognize that this is currently biased towards our own use cases and experiences, but we hope to engage in robust discussion with the open source community to build a comprehensive resource.
```

File: content/03.introduction.md [hunk 5]
```diff
@@ -1,29 +1,18 @@
 ## Introduction
+*(draft)*
 
+Sequencing data are now widely available for species across the tree of life, and new sequencing data continues to be generated at an incredible rate [@url:https://www.ncbi.nlm.nih.gov/sra/docs/sragrowth/].
+The wealth of information present in high-throughput sequencing data has already revolutionized our understanding of the diversity and function of communities, building basic understanding from ecosystems to human health.
```
Comment (taylorreiter) [inline content/03.introduction.md]:
```suggestion
The wealth of information present in high-throughput sequencing data has already revolutionized our understanding of the diversity and function of organisms and communities, building basic understanding from ecosystems to human health.
```

File: content/03.introduction.md [hunk 6]
```diff
@@ -1,29 +1,18 @@
...
-These tools enable researchers to build reproducible analyses that can be automatically executed in a directed fashion.
-With integrated installations, these workflows can work across different computational systems, and can even serve as a form of documentation for the analysis.
-Finally, when paired with new tools leveraging computational approximations, this suite of tools enables researchers to cope with the enormity of sequencing data.
-%have emerged a promising solution to coping with the enormity of sequencing data.
-%..provide researchers a framework/structure
-%Adopting workflow-based systems may be the single best step you can take to improve your analyses (here's where to talk up workflows!)
-%bonus: these integrate with software installation! Also provide a bunch of other neat data-sciencey logging and benchmarking.
+Since the latest paper in 2017 [@doi:10.1371/journal.pcbi.1005510],  key advancements in workflow scripting, software management, and tools that handle biological data at scale have vastly increased the power of analysis management.
```
Comment (taylorreiter) [inline content/03.introduction.md:9]: love this sentence!

File: content/04.workflows-and-software.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-## Workflows and Software Management
+## Workflows facilitate data-intensive biology
```
Comment (taylorreiter) [inline content/04.workflows-and-software.md:1]: Ya they do!!! Love this header

File: content/04.workflows-and-software.md [hunk 8]
```diff
@@ -89,7 +88,8 @@ This ensures that an environment is completely reproducible, and is common for p
...
-also specifies which Conda channels to search and download programs from. When specified in a YAML file, conda environments are easily transferable between computers and operating systems. Further, because the version of each package installed in an environment is recorded, workflow reproducibility is enhanced. Although portions of conda may be superceded by alternative solutions [@url:https://github.com/QuantStack/mamba], this model of software installation and management will likely remain.](images/conda_figure.svg){#fig:conda_figure}
+also specifies which Conda channels to search and download programs from. When specified in a YAML file, conda environments are easily transferable between computers and operating systems. Further, because the version of each package installed in an environment is recorded, workflow reproducibility is enhanced. Although portions of conda may be superceded by alternative solutions [@url:https://github.com/QuantStack/mamba], this model of software installation and management will likely
```
Comment (taylorreiter) [inline content/04.workflows-and-software.md:91]: I love this last sentence!

File: content/05.project-management.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
+## Workflow-Based Project Management
+
+Computational project management is a learned skill that takes time to implement.
+Biological analyses often span hundreds of steps and involve a myriad of decisions ranging from small-scale tool and parameter choices to larger-scale decisions around computational experimental design and statistical analyses.
+It is rare (if not impossible) to build or find a workflow that to analyze your data from start to finish without testing, troubleshooting, and iteration.
```
Comment (taylorreiter) [inline content/05.project-management.md]:
```suggestion
It is rare (if not impossible) to build or find a workflow that analyzes your data from start to finish without testing, troubleshooting, and iteration.
```

File: content/05.project-management.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+Fortunately, with a little direction, workflow systems both simplify and improve computational project management.
+
+### Systematically document your workflows
```
Comment (taylorreiter) [inline content/05.project-management.md:8]: This paragraph doesn't really work as is...it doesn't set up for the information to come in the next few paragraphs, and doesn't provide strategies to combat the ways in which workflow self-documentation is limited.

File: content/05.project-management.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+If a key piece of a workflow inexplicably stops working, good version control can allow you to rewind in time and identify differences from when the pipeline worked to when it stopped working.
+
+![**Version Control** Version control systems (e.g. git, mercurial) work by storing incremental differences in files from one saved version ("commit") to the next. To visualize the differences between each version, text editors such as Atom and online services such as GitHub, GitLab and Bitbucket use red highlight to denote deletions, and green highlighting to denote additions. In this trivial example, a typo in version 1 (in red) was corrected (green). These systems are extremely useful for working code and manuscript development, as it is possible to return to the snapshot of any saved version. This means that version control systems save you from accidental deletions, preserve code you thought you no longer needed, and preserve a record of project change over time.](images/version_control.svg){#fig:version_control}
```
Comment (taylorreiter) [inline content/05.project-management.md]:
```suggestion
![**Version Control** Version control systems (e.g. Git, Mercurial) work by storing incremental differences in files from one saved version ("commit") to the next. To visualize the differences between each version, text editors such as Atom and online services such as GitHub, GitLab and Bitbucket use red highlight to denote deletions, and green highlighting to denote additions. In this trivial example, a typo in version 1 (in red) was corrected (green). These systems are extremely useful for code and manuscript development, as it is possible to return to the snapshot of any saved version. This means that version control systems save you from accidental deletions, preserve code you thought you no longer needed, and preserve a record of project changes over time.](images/version_control.svg){#fig:version_control}
```

File: content/05.project-management.md [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+Plotting tools like plotly and vega-lite produce single interactive plots that can be shared with collaborators or integrated into websites [@url:https://plotly.com/; @doi:10.1109/TVCG.2016.2599030].
+Interactive visualizations are also helpful in exploratory data analysis.
+](images/interactive_viz.png){#fig:interactiveviz}
+
+
+### Getting started developing workflows
+
+In our experience, the best way to have your workflow system work _for_ you is to include as much of your analysis as possible within the automated workflow framework, use self-documenting names, include analysis visualizations, and keep rigorous documentation alongside that would enable you to understand each decision and entirely reproduce any manual steps. Some of the tools discussed above will inevitably change over time, but these principles apply broadly and will help you design clear, well-documented, and reproducible analyses. Ultimately, you'll need to experiment with the ways that work for you -- what is most important is to develop a clear set of strategies and implement them tenaciously. Here are a few practical tips to get started:
```
Comment (taylorreiter) [inline content/05.project-management.md]:
```suggestion
In our experience, the best way to have your workflow system work _for_ you is to include as much of your analysis as possible within the automated workflow framework, use self-documenting names, include analysis visualizations, and keep rigorous documentation alongside your workflow that enables you to understand each decision and entirely reproduce any manual steps. 
Some of the tools discussed above will inevitably change over time, but these principles apply broadly and will help you design clear, well-documented, and reproducible analyses. 
Ultimately, you'll need to experiment with the ways that work for you -- what is most important is to develop a clear set of strategies and implement them tenaciously. 
Here are a few practical tips to get started.
```

File: content/05.project-management.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+
+**Document your process** Write documentation in plain text, so your notes files can be saved under version control with your workflow. We recommend using Markdown language for your notes so they can include helpful headings, code formatting, and embedded images. We recommend using software with visual text previewing such as Hackmd or Atom [@url:https://hackmd.io/, @url:https://atom.io/].
+
+**Develop your workflow**
+Now that you have the workflow running on new data, you can begin to modify a workflow step to meet your needs.
+Run each modified step on your subsampled data, checking that the output files look correct.
+Keep the tool documentation and tutorials on hand; as with any language, remembering workflow-specific syntax takes time and practice.
+
+**Back up early and often**
+Now that you've written new code, the first thing to do is back up your changes in an online repository such as GitHub, GitLab, or Bitbucket.
```
Comment (taylorreiter) [inline content/05.project-management.md]:
```suggestion
As you write new code, back up your changes in an online repository such as GitHub, GitLab, or Bitbucket.
```

File: content/05.project-management.md [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+Turn on this reporting according to workflow documentation and record the resources required for running each tool on a full-size single sample.
+Use these estimates to set appropriate resource limits for each step when executing the workflow on your remaining samples.
+
+**Find a community and ask for help when you need it**
+If there are local user groups for your workflow language, try to get involved with the community.
+When you are first learning, help from more advanced users can save you hours of frustration.
+After you've progressed, providing that same help to new users can help you cement the syntax in your mind and tackle more advanced uses.
+If no local community is available, look online for forums related to your workflow language.
+These workflow systems have been enthusiastically adopted by the open science community, and as a consequence, there is a critical mass of tutorials and open access code, code discussion on forums and via social media, particularly twitter.
+Be respectful of people's time and energy: post in the relevant workflow forums only when you've hit a stopping point you're unable to work through.
```
Comment (taylorreiter) [inline content/05.project-management.md]:
```suggestion
Be respectful of people's time and energy: post in the relevant workflow forums only when you have hit a stopping point you are unable to work through.
```

File: content/06.data-resource-management.md [hunk 15]
```diff
@@ -1,35 +1,42 @@
...
 Finally, rather than focusing on certain data types or organisms, some repositories are designed to hold any data and metadata associated with a specific project or manuscript (e.g. Open Science Framework, Dryad, Zenodo [@doi:10.5195/JMLA.2017.88]).
 
-### Generating your own data
+#### Consider analysis when generating your own data
 
 If generating your own data, proper experimental design and planning are essential.
 For cost-intensive sequencing data, there are a range of decisions about experimental design and sequencing (including sequencing type, sequencing depth per sample, and biological replication) that impact your ability to properly address your research question.
 These considerations will be different for different types of sequence analysis.
 While we have curated a series of domain-specific references that may be useful as go about designing your experiment (see **Table @tbl:seq_resources**), conducting discussions with experienced bioinformaticians and statisticians, **prior to beginning your experiments** if possible, is the best way to ensure you will have sufficient statistical power to detect effects.
```
Comment (taylorreiter) [inline content/06.data-resource-management.md]:
```suggestion
While we have curated a series of domain-specific references that may be useful as you go about designing your experiment (see **Table @tbl:seq_resources**), conducting discussions with experienced bioinformaticians and statisticians, **prior to beginning your experiments** if possible, is the best way to ensure you will have sufficient statistical power to detect effects.
```

File: content/06.data-resource-management.md [hunk 16]
```diff
@@ -44,93 +51,50 @@ Given the resources invested in collecting samples for sequencing, it's importan
...
+](images/checksum.svg){#fig:checksum height=2.5in}
 
-Workflow sysems allow users to generate an automatable, repeatable workflow.
-This removes the necessity of storing some intermediate files as these can be easily regenerated.
-However, storing or backing up raw data or computationally intensive results can be a burden in data intensive workflows.
-Many universities provide cloud storage space (e.g. through google drive, box, dropbox etc), and researchers can pay for individual storage on these services, or services attached to cloud computing (e.g. Amazon Web Services).
-Full computer backups can be conducted to these storage locations (e.g. with rclone [@doi:10.1111/2041-210X.12550]), or there are also a number of paid services that will conduct backups at regular intervals (e.g. Backblaze, Dropbox Pro).
-Free online repositories mentioned in the version control section like OSF, Dryad, and Zenodo can also store or backup important files.
+### Getting started: strategies for quality control
```
Comment (taylorreiter) [inline content/06.data-resource-management.md:85]: This is a really abrupt transition..."Getting started" comes out of no where

File: content/06.data-resource-management.md [hunk 17]
```diff
@@ -144,25 +108,99 @@ Parsing and visualizing log files with a tool like MultiQC can improve interpret
 |  cut | cut columns from a table | cut -d"," -f1 samples.csv |
 Table: Some bash commands are useful to quickly explore the contents of a file. By using these commands, the user can detect common formatting problems or other abnormalities. {#tbl:bash_commands}
 
+**Visualize your data**
+Visualization is another powerful way to pick out unusual or unexpected patterns.
+Although large abnormalities may be clear from looking at files, others may be small and difficult to find.
+Visualizing raw sequencing data with FastQC (**Figure {@fig:multiqc}B**) and processed sequencing data with tools like the Integrative Genome Viewer and plotting tabular results files using python or R can make aberrant or inconsistent results easier to track down [@url:https://www.bioinformatics.babraham.ac.uk/projects/fastqc/; @doi:10.1093/bib/bbs017].
```
Comment (taylorreiter) [inline content/06.data-resource-management.md]:
```suggestion
Visualizing raw sequencing data with FastQC (**Figure {@fig:multiqc}A**) and processed sequencing data with tools like the Integrative Genome Viewer and plotting tabular results files using python or R can make aberrant or inconsistent results easier to track down [@url:https://www.bioinformatics.babraham.ac.uk/projects/fastqc/; @doi:10.1093/bib/bbs017].
```

File: content/06.data-resource-management.md [hunk 18]
```diff
@@ -144,25 +108,99 @@ Parsing and visualizing log files with a tool like MultiQC can improve interpret
...
+FastQC measures and summarizes 10 quality metrics and provides recommendations for whether the sample is within an acceptable quality range.
+Not all metrics readily apply to all sequencing data types. For example, while multiple GC peaks might be concerning in whole genome sequencing of a bacterial isolate, we would expect a non-normal distrubtion for some metagenome samples that contain organisms with diverse GC content.
+Samples like this can be seen in red in this figure.
 **B** MultiQC summary of Salmon *quant* reads mapped per sample for RNA-seq samples [@doi:10.1038/nmeth.4197]. MultiQC finds and automatically parses log files from other tools and generates a combined report and parsed data tables that include all samples. MultiQC currently supports 88 tools. In this figure, we see that MultiQC summarizes the number of reads mapped and percent of reads mapped, two values that are reported in the Salmon log files.
 ](images/multiqc.svg){#fig:multiqc}
 
+**Pay attention to warnings and logging**
```
Comment (taylorreiter) [inline content/06.data-resource-management.md]:
```suggestion
**Pay attention to warnings and log files**
```

File: content/06.data-resource-management.md (hunk 16 above)
Comment (taylorreiter) [reply content/06.data-resource-management.md:85]: I think the section flows though, maybe just needs a different header title?

Comment (taylorreiter) [reply content/06.data-resource-management.md:85]: Maybe to mirror the next section, "Getting started with data quality control"

File: content/06.data-resource-management.md [hunk 19]
```diff
@@ -144,25 +108,99 @@ Parsing and visualizing log files with a tool like MultiQC can improve interpret
...
+|  Open Science Data Cloud | Research allocation |  |
+|  Wasabi | Paid | data storage solution only |
+Table: **Research cloud resources** Cloud provider indicates the name of the cloud, standard model indicates the most common route toward using the cloud, and limitations indicates limitations in access or services provided by the cloud. {#tbl:computational_resources}
+
+### Getting started with resource management
+
+As the scale of data increases, the resources required for analysis can balloon. Bioinformatic workflows can be long-running, require high-memory systems, or involve intensive file manipulation. Some of the strategies below may help you manage computational resources for your project.
+
+**Apply for research units if eligible**
+There are a number of cloud computing services that offer grants providing computing resources to data-intensive researchers {@tbl:computational_resources}. In some cases, the resources provided may be sufficient to cover your entire analysis.
```
Comment (taylorreiter) [inline content/06.data-resource-management.md]:
```suggestion
There are a number of cloud computing services that offer grants providing computing resources to data-intensive researchers (**Table {@tbl:computational_resources}**). In some cases, the resources provided may be sufficient to cover your entire analysis.
```

File: content/04.workflows-and-software.md [hunk 20]
```diff
@@ -89,7 +88,8 @@ This ensures that an environment is completely reproducible, and is common for p
 **A. Conda Recipe Repositories** Each program distributed via Conda has a "recipe" describing all software dependencies needed for Conda installation (each of which must also be installable via conda). These are stored and managed in separate "channels", some of which specialize (e.g. "bioconda" specializes in bioinformatic software, "r" specializes in R language packages) [@doi:10.1038/s41592-018-0046-7]. **B. Use Conda Environments to Avoid Installation Conflicts**  Conda does not require root
 privileges for software installation, thus enabling use by researchers working on shared cluster systems. However, even user-based software installation can encounter dependency conflicts. For example, you might need to use python2 to install and run a program (e.g. older scripts written by members of your lab), while also using snakemake to execute your workflows (requires python3.5). By installing each program into an isolated "environment" that contains only the software required to run that
```
Comment (taylorreiter) [inline content/04.workflows-and-software.md]:
```suggestion
privileges for software installation, thus enabling use by researchers working on shared cluster systems. However, even user-based software installation can encounter dependency conflicts. For example, you might need to use python2 to install and run a program (e.g. older scripts written by members of your lab), while also using snakemake to execute your workflows (requires python>=3.5). By installing each program into an isolated "environment" that contains only the software required to run that
```

**Review by taylorreiter — APPROVED** [review]
Comments on PR:
+ Author summary will change after we receive input/feedback from the community?
+ We no longer reference plotly in the text, just have a viz of it. Not sure if this is a problem or not!
+ The section "Getting started: strategies for quality control" comes out of no where
+ Fix intro paragraph in "Systematically document your workflows" 

Typos etc that need to change (github didn't just let me suggest these things):
+ "When paired with proper software management, fully-contained workflows are scaleable, robust to software updates, and executable across platforms, meaning they will likely still execute the same set of commands with little investment by the user in weeks, months, or years from the time of writing." -- do we need "likely" in this sentence?
+ Figure 1: 'facilitate/streamline/simplify'  should become "facilitates and simiplifies"
+ Figure 1, B: (e.g. conda, singularity) should be (e.g. conda, singularity, docker)
+ Figure 1, D: missing a space
+ In "Choosing a workflow system":  "These tools are specified in an underlying workflow language, but are packaged in a more user-friendly manner command-line script that coordinates and executes the workflow." --> "These tools are specified in an underlying workflow language, but are packaged as a user-friendly command-line script that coordinates and executes the workflow"
+ in "Wrangling scientific software": "These tools are heterogeneous in nature: written by researchers working in different coding languages, with varying types software design and optimization, and often for specific analysis goals." --> "These tools are heterogeneous in nature: written by researchers working in different coding languages, with varied strategies for software design and optimization, and often for specific analysis goals."
+ "On many systems, system-wide software management is overseen by system administrators, who ensure commonly-used software is installed into a “module” system that is available to all users." 
+ "Getting started with software management": "If the workflow is then executed with the “–use-conda”, “–use-singularity”, or “-use-envmodules” options, the specified environment or container will be downloaded and installed (or system module will be loaded), used to run that analysis step, and stored for future analysis runs." --> "If the workflow is then executed with specific flags, the specified environment or container will be downloaded and installed (or system module will be loaded), used to run that analysis step, and stored for future analysis runs."

**Pull request merged**